metadata-ingestion/src/datahub/ingestion/source_report/ingestion_stage.py

```diff
@@ -20,31 +23,70 @@
 PROFILING = "Profiling"
 
 
+class IngestionHighStage(Enum):
+    """
+    The high-level stages at the framework level
+    Team to add more stages as needed
+    """
+
+    PROFILING = "Profiling"
+    _UNDEFINED = "Ingestion"
+
+
 @dataclass
 class IngestionStageReport:
-    ingestion_stage_durations: TopKDict[str, float] = field(default_factory=TopKDict)
+    ingestion_high_stage_seconds: dict[str, float] = field(
+        default_factory=lambda: defaultdict(float)
+    )
+    ingestion_stage_durations: TopKDict[Tuple[str, str], float] = field(
+        default_factory=TopKDict
+    )
 
-    def new_stage(self, stage: str) -> "IngestionStageContext":
-        return IngestionStageContext(stage, self)
+    def new_stage(
+        self, stage: str, high_stage: IngestionHighStage = IngestionHighStage._UNDEFINED
+    ) -> "IngestionStageContext":
+        return IngestionStageContext(stage, self, high_stage)
+
+    def new_high_stage(self, stage: IngestionHighStage) -> "IngestionStageContext":
+        return IngestionStageContext("", self, stage)
```

Comment:
why not only one method?

```suggestion
    def new_stage(
        self, stage: str, high_stage: IngestionHighStage = IngestionHighStage._UNDEFINED
    ) -> "IngestionStageContext":
        return IngestionStageContext(stage, self, high_stage)
```

Reply:
To keep our code clean. We will have to keep on doing this 
```
with self.get_report().new_stage("", IngestionHighStage.PROFILING):
```

everywhere which is quite odd to use everywhere.

Reply:
I don't understand the issue.

Anyway, IMO eventually we should have just one method: 

```
# Either: if you want abitlity to track stuff at high_stage with no nested stage
new_stage(high_stage: IngestionHighStage = _UNDEFINED, stage: Optional[str] = None)
# or: make nested stage required
new_stage(high_stage: IngestionHighStage = _UNDEFINED, stage: Optional[str] = None)
```

Actually, `_UNDEFINED` value should also be valid during the transition, eventually all ingestors should use proper values and different from that one.

We could have/keep other variations just to make easier the transition, but we could set those other variations as `deprecated`.